compiler/src/iree/compiler/Codegen/ExternalInterfaces/GPUEncodingExternalModels.cpp

```diff
@@ -21,17 +22,22 @@
 
 #include "iree/compiler/Codegen/ExternalInterfaces/GPUEncodingExternalModels.h"
 
-#include <cfloat>
-
-#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenTypes.h"
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.h"
 #include "iree/compiler/Codegen/Dialect/Codegen/Utils/Utils.h"
 #include "iree/compiler/Codegen/Dialect/GPU/IR/GPUTileSwizzleUtils.h"
 #include "iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.h"
 #include "iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUDialect.h"
 #include "iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUOps.h"
 #include "iree/compiler/Dialect/Encoding/IR/EncodingOps.h"
+#include "iree/compiler/Dialect/Encoding/IR/EncodingTypes.h"
 #include "llvm/Support/Debug.h"
 #include "mlir/Dialect/Linalg/IR/LinalgInterfaces.h"
+#include "mlir/IR/BuiltinAttributes.h"
+
+#include <cassert>
+#include <cfloat>
+#include <cstdint>
+#include <numeric>
```

Comment:
I don't remember which include style we are following. LLVM puts the system headers at the end, while Google puts them in the middle. IREE is following the mixed styles. It looks okay to me because our Codegen codebase is very close to MLIR codebase.

@benvanik @MaheshRavishankar which style should we follow?

https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes
https://llvm.org/docs/CodingStandards.html#include-style

Reply:
We should write it down somewhere on iree.dev. I remember the include order is something that we explicitly tweaked when switching to the llvm clang-format style: https://github.com/iree-org/iree/blob/main/compiler/.clang-format